pkg/coordinator/sweep.go

```diff
@@ -127,3 +132,129 @@ func ValidateDepositString(depositString string) error {
 	}
 	return nil
 }
+
+func EstimateDepositsSweepFee(
+	tbtcChain tbtc.Chain,
+	btcChain bitcoin.Chain,
+	depositsCount int,
+) error {
+	_, _, perDepositMaxFee, _, err := tbtcChain.GetDepositParameters()
+	if err != nil {
+		return fmt.Errorf("cannot get deposit tx max fee: [%v]", err)
+	}
+
+	fees := make(map[int]int64)
+	var depositsCountKeys []int
+
+	if depositsCount > 0 {
+		depositsCountKeys = append(depositsCountKeys, depositsCount)
+	} else {
+		sweepMaxSize, err := tbtcChain.GetDepositSweepMaxSize()
+		if err != nil {
+			return fmt.Errorf("cannot get sweep max size: [%v]", sweepMaxSize)
+		}
+
+		for i := 1; i <= int(sweepMaxSize); i++ {
+			depositsCountKeys = append(depositsCountKeys, i)
+		}
+	}
+
+	for _, depositsCountKey := range depositsCountKeys {
+		fee, err := estimateDepositsSweepFee(
+			btcChain,
+			depositsCountKey,
+			perDepositMaxFee,
+		)
+		if err != nil {
+			return fmt.Errorf(
+				"cannot estimate fee for deposits count [%v]: [%v]",
+				depositsCountKey,
+				err,
+			)
+		}
+
+		fees[depositsCountKey] = fee
+	}
+
+	err = printDepositsSweepFeeTable(fees)
+	if err != nil {
+		return fmt.Errorf("cannot print fees table: [%v]", err)
+	}
+
+	return nil
+}
+
+func estimateDepositsSweepFee(
+	btcChain bitcoin.Chain,
+	depositsCount int,
+	perDepositMaxFee uint64,
+) (int64, error) {
+	transactionSize, err := bitcoin.NewTransactionSizeEstimator().
+		// 1 P2WPKH main UTXO input.
+		AddPublicKeyHashInputs(1, true).
+		// depositsCount P2WSH deposit inputs.
+		AddScriptHashInputs(depositsCount, depositScriptByteSize, true).
+		// 1 P2WPKH output.
+		AddPublicKeyHashOutputs(1, true).
+		VirtualSize()
+	if err != nil {
+		return 0, fmt.Errorf("cannot estimate transaction virtual size: [%v]", err)
+	}
+
+	feeEstimator := bitcoin.NewTransactionFeeEstimator(btcChain)
+
+	fee, err := feeEstimator.EstimateFee(transactionSize)
+	if err != nil {
+		return 0, fmt.Errorf("cannot estimate transaction fee: [%v]", err)
+	}
+
+	// Compute the maximum possible fee for the entire sweep transaction.
+	maxFee := uint64(depositsCount) * perDepositMaxFee
+	// Make sure the proposed fee does not exceed the maximum possible fee.
+	fee = int64(math.Min(float64(fee), float64(maxFee)))
+
+	return fee, nil
+}
+
+func printDepositsSweepFeeTable(fees map[int]int64) error {
+	writer := tabwriter.NewWriter(
+		os.Stdout,
+		2,
+		4,
+		1,
+		' ',
+		tabwriter.AlignRight,
+	)
+
+	_, err := fmt.Fprintf(writer, "deposits count\tfee (satoshis)\t\n")
```

Comment:
We could make it clear in the table that this is the fee for the entire sweep transaction. It is pretty unclear to someone not familiar with this code they can just copy&paste the value to propose the deposit sweep.

This is what is currently printed:
```
 deposits count fee (satoshis)
             20           1935
```
IMO it is unclear if it is per-deposit or per deposits count fee.

This is the documentation of the deposit sweep proposal command:
```
--fee int         fee for the entire bitcoin transaction (satoshi)
```


```suggestion
	_, err := fmt.Fprintf(writer, "deposits count\ttotal fee (satoshis)\t\n")
```

Reply:
Good point, see 3239f5d7